### PR TITLE
Add r-baseset recipe

### PR DESCRIPTION
**Pull request by mjsteinbaugh**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mjsteinbaugh** [comment]
I ran this locally and it builds successfully on osx_64

**Comment by mjsteinbaugh** [comment]
@conda-forge/help-r This is a CRAN package not currently available on conda-forge

File: recipes/r-baseset/build.sh
Comment (mfansler) [inline recipes/r-baseset/build.sh:34]: This antiquated boilerplate; contemporary is simply:

```suggestion
export DISABLE_AUTOBREW=1
${R} CMD INSTALL --build . ${R_ARGS}
```

File: recipes/r-baseset/bld.bat
Comment (mfansler) [inline recipes/r-baseset/bld.bat:2]: Use contemporary boilerplate:

```suggestion
"%R%" CMD INSTALL --build . %R_ARGS%
IF %ERRORLEVEL% NEQ 0 exit /B 1
```

File: recipes/r-baseset/meta.yaml
Comment (mfansler) [inline recipes/r-baseset/meta.yaml:25]: Preemptively including `cross-r-base`:

```suggestion
    - cross-r-base {{ r_base }}  # [build_platform != target_platform]
    - {{ posix }}zip             # [win]
```

File: recipes/r-baseset/meta.yaml
Comment (mfansler) [inline recipes/r-baseset/meta.yaml:43]: Defer to the upstream URLs found in DESCRIPTION:

```suggestion
  home: https://docs.ropensci.org/BaseSet/
  dev_url: https://github.com/ropensci/BaseSet
```

File: recipes/r-baseset/meta.yaml
Comment (mfansler) [inline recipes/r-baseset/meta.yaml:45]: Use `Description:` text and declare `license_family`:

```suggestion
  summary: Implements a class and methods to work with sets, doing intersection, union, complementary
    sets, power sets, cartesian product and other set operations in a "tidy" way. These
    set operations are available for both classical sets and fuzzy sets. Import sets
    from several formats or from other several data structures.
  license_family: MIT
```

File: recipes/r-baseset/meta.yaml
Comment (mfansler) [inline recipes/r-baseset/meta.yaml:48]: CRAN packages with MIT always include an additional license file:

```suggestion
    - LICENSE

```

File: recipes/r-baseset/meta.yaml
Comment (mfansler) [inline recipes/r-baseset/meta.yaml:53]: AFAIK, we don't typically allow organizations to have maintainer rights. Please explain if you believe an exception is justified.

```suggestion
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for the submission! ❤️ 

I've suggested several changes. Note that most of these are a matter of conforming to the standards implemented in [`conda_r_skeleton_helper`](https://github.com/bgruening/conda_r_skeleton_helper). In the future, please consider using that tool to generate recipes for CRAN packages.

**Review by mjsteinbaugh — COMMENTED** [review]
@mfansler Thanks for the help with the recipe. I fixed a typo in the archive tarball URL as well. Think it looks OK to merge?

**Review by mfansler — APPROVED** [review]

**Pull request merged**
